### PR TITLE
Debit card auto reconciliation

### PR DESCRIPTION
**Pull request by ibrechin**

### CODE REVIEW

File: mtp_api/apps/payment/views.py [hunk 1]
```diff
@@ -1,14 +1,38 @@
...
+class BatchView(
+    mixins.ListModelMixin, viewsets.GenericViewSet
+):
+    queryset = Batch.objects.all()
+    serializer_class = BatchSerializer
+    filter_backends = (filters.DjangoFilterBackend,)
+    filter_class = BatchListFilter
+
+    permission_classes = (
+        IsAuthenticated, ActionsBasedPermissions, BankAdminClientIDPermissions
```
Comment (ushkarev) [inline mtp_api/apps/payment/views.py]: i think we need a subclass of `ActionsBasedPermissions` which includes 'view_batch' in the 'list' `actions_perms_map`

File: mtp_api/apps/payment/managers.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+        update_set = self.get_queryset().filter(
+            status=PAYMENT_STATUS.TAKEN,
+            credit__received_at__gte=set_worldpay_cutoff(start_date),
+            credit__received_at__lt=set_worldpay_cutoff(end_date),
+            credit__reconciled=False
+        ).select_for_update()
+
+        if update_set.count() > 0:
+            batch_model = apps.get_model('payment', 'batch')
```
Comment (ushkarev) [inline mtp_api/apps/payment/managers.py]: probably simpler to have a local import of the model

**Comment by ibrechin** [comment]
retest

**Comment by ibrechin** [comment]
retest this

**Comment by ibrechin** [comment]
retest this please

**Pull request merged**